src/main/java/edu/illinois/cs/dt/tools/fixer/CleanerFixerPlugin.java

```diff
@@ -0,0 +1,211 @@
+        try (final Stream<Path> paths = Files.walk(Paths.get(project.getBuild().getTestSourceDirectory()))) {
+            paths.filter(Files::isRegularFile)
+                    .forEach(testFiles::add);
+        }
+
+        // TODO: Remove these and read from minimized files
+        final String polluterTestName = System.getProperty("dt_fixer.polluter");
+        final String cleanerTestName = System.getProperty("dt_fixer.cleaner");
+        final String victimTestName = System.getProperty("dt_fixer.victim");
+
+        final Optional<JavaMethod> cleanerMethodOpt = JavaMethod.find(cleanerTestName, testFiles, classpath);
+        final Optional<JavaMethod> victimMethodOpt = JavaMethod.find(victimTestName, testFiles, classpath);
+
+        if (!cleanerMethodOpt.isPresent()) {
+            TestPluginPlugin.error("Could not find method " + cleanerTestName);
+            TestPluginPlugin.error("Tried looking in: " + testFiles);
+            return;
+        }
+
+        if (!victimMethodOpt.isPresent()) {
+            TestPluginPlugin.error("Could not find method " + victimTestName);
+            TestPluginPlugin.error("Tried looking in: " + testFiles);
+            return;
+        }
+
+        applyFix(polluterTestName, cleanerMethodOpt.get(), victimMethodOpt.get());
+    }
+
+    private void backup(final JavaFile javaFile) throws IOException {
+        final Path path = CleanerPathManager.backupPath(javaFile.path());
+        Files.copy(javaFile.path(), path, StandardCopyOption.REPLACE_EXISTING);
+    }
+
+    // TODO: Extract this logic out to a more generalized fixer that is separate, so we can reuse it
+    // TODO: for cleaners, santa clauses, etc.
+    private void applyFix(final String polluterName,
+                          final JavaMethod cleanerMethod,
+                          final JavaMethod victimMethod) throws Exception {
+        backup(victimMethod.javaFile());
+
+        // Compile the test without our fix
+//        tryRecompile(victimMethod.javaFile());
+
+        runMvnInstall();
+
+        // Check if we pass in isolation before fix
+        // TODO: We should get rid of this check eventually, because it won't work for the case of \santaClause
+        TestPluginPlugin.info("Running victim test in isolation without code from cleaner.");
+        if (!testOrderPasses(Collections.singletonList(victimMethod.methodName()))) {
+            TestPluginPlugin.error("Test fails in isolation");
+            return;
+        }
+
+        // Do our fix
+        TestPluginPlugin.info("Applying code from cleaner and recompiling.");
+        final NodeList<Statement> cleanerStmts = cleanerMethod.body().getStatements();
+
+        // Note: this modifies victimMethod, so when we eventually make it delta-debug, we will want
+        // to make a copy of the victimMethod somewhere
+        victimMethod.prepend(cleanerStmts);
+        victimMethod.javaFile().writeAndReloadCompilationUnit();
```

Comment:
I had some issue in the original code concerning the rewriting and recompiling, where I had to make a new JavaFile there each time to properly write out after every change. I don't know if such an issue will happen here.